userspace/libsinsp/sinsp.h

```diff
@@ -1273,12 +1272,59 @@ VISIBILITY_PRIVATE
 	// *	user added/removed events
 	// * 	group added/removed events
 	// *    async events produced by sinsp or plugins
-#ifndef __EMSCRIPTEN__
-	tbb::concurrent_queue<std::shared_ptr<sinsp_evt>> m_pending_state_evts;
-#endif
+
+	// m_injected_evts comparator
+	using sinsp_evt_ptr = std::unique_ptr<sinsp_evt>;
+	struct state_evts_less
+	{
+		bool operator()(const sinsp_evt_ptr& l, const sinsp_evt_ptr& r)
+		{
+			return l->get_ts() != static_cast<uint64_t>(-1) && l->get_ts() > r->get_ts();
+		}
+	};
+
+	// priority queue to hold injected events
+	mpsc_priority_queue<sinsp_evt_ptr, state_evts_less> m_pending_state_evts;
 
 	// Holds an event dequeued from the above queue
-	std::shared_ptr<sinsp_evt> m_state_evt;
+	sinsp_evt_ptr m_state_evt;
```

Comment:
can we rename it to `async_evt`? `state_evt` is quite generic
```suggestion
	sinsp_evt_ptr m_async_evt;
```